### PR TITLE
MINOR: Fix Scala 2.13 compiler warnings

### PR DESCRIPTION
**Pull request by ijuma**
Once Scala 2.13.2 is officially released, I will submit a follow up PR
that enables `-Xfatal-warnings` with the necessary warning
exclusions. Compiler warning exclusions were only introduced in 2.13.2
and hence why we have to wait for that. I used a snapshot build to
test it in the meantime.

Changes:
* Remove Deprecated annotation from internal request classes
* Class.newInstance is deprecated in favor of
Class.getConstructor().newInstance
* Replace deprecated JavaConversions with CollectionConverters
* Remove unused kafka.cluster.Cluster
* Don't use Map and Set methods deprecated in 2.13:
    - collection.Map +, ++, -, --, mapValues, filterKeys, retain
    - collection.Set +, ++, -, --
* Add scala-collection-compat dependency to streams-scala and
update version to 2.1.4.
* Replace usages of deprecated Either.get and Either.right
* Replace usage of deprecated Integer(String) constructor
* `import scala.language.implicitConversions` is not needed in Scala 2.13
* Replace usage of deprecated `toIterator`, `Traversable`, `seq`,
`reverseMap`, `hasDefiniteSize`
* Replace usage of deprecated alterConfigs with incrementalAlterConfigs
where possible
* Fix implicit widening conversions from Long/Int to Double/Float
* Avoid implicit conversions to String
* Eliminate usage of deprecated procedure syntax
* Remove `println`in `LogValidatorTest` instead of fixing the compiler
warning since tests should not `println`.
* Eliminate implicit conversion from Array to Seq
* Remove unnecessary usage of 3 argument assertEquals
* Replace `toStream` with `iterator`
* Do not use deprecated SaslConfigs.DEFAULT_SASL_ENABLED_MECHANISMS
* Replace StringBuilder.newBuilder with new StringBuilder
* Rename AclBuffers to AclSeqs and remove usage of `filterKeys`
* More consistent usage of Set/Map in Controller classes: this also fixes
deprecated warnings with Scala 2.13
* Add spotBugs exclusion for inliner artifact in KafkaApis with Scala 2.12.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cmccabe** [comment]
Thanks for this.  Much better!

It looks like spotbugs found something.

> Remove test `println`

Why is this one needed?

**Comment by ijuma** [comment]
Yes, I am investigating the spotBugs issue and the test failures.

I removed the `println` from the test because we don't generally do that, right? It was triggering a warning and I decided to remove it altogether instead of fixing the warning. Any reason to keep it?

**Comment by ijuma** [comment]
Scala 2.12 passed, 2 flaky failures with Scala 2.13:

* kafka.server.AlterReplicaLogDirsRequestTest > testAlterReplicaLogDirsRequestErrorCode
* org.apache.kafka.streams.integration.QueryableStateIntegrationTest > concurrentAccesses

**Comment by ijuma** [comment]
retest this please

File: core/src/main/scala/kafka/log/LogCleanerManager.scala [hunk 1]
```diff
@@ -410,7 +410,7 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
         if (checkpoint != null) {
           val existing = checkpoint.read()
           if (existing.getOrElse(topicPartition, 0L) > offset)
-            checkpoint.write(existing + (topicPartition -> offset))
+            checkpoint.write(mutable.Map() ++= existing += topicPartition -> offset)
```
Comment (chia7712) [inline core/src/main/scala/kafka/log/LogCleanerManager.scala:413]: Pardon me, I failed to catch the point of this change.

**Review by chia7712 — APPROVED** [review]
Should we disable to import ```scala.collection.JavaConverters``` to scala file?  If so, we should add the rule to import-control file.

**Comment by ijuma** [comment]
@chia7712 My follow up change will effectively do that since the Scala compiler will fail compilation with Scala 2.13 if you try to use it.

File: core/src/main/scala/kafka/log/LogCleanerManager.scala (hunk 1 above)
Comment (ijuma) [reply core/src/main/scala/kafka/log/LogCleanerManager.scala:413]: `scala.collection.Map.+` is deprecated.

**Comment by ijuma** [comment]
A different flaky failure for Scala 2.13, 2.12 passed:

> kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup

So, this should be good to merge pending a successful review.

**Review by omkreddy — APPROVED** [review]
@ijuma Thanks for the PR. LGTM.

**Pull request merged**